Zend/zend_gc.c

```diff
@@ -365,13 +373,56 @@ ZEND_API void ZEND_FASTCALL gc_remove_from_buffer(zend_refcounted *ref)
 	}
 }
 
-static void gc_scan_black(zend_refcounted *ref)
+static gc_refcounted_stack* gc_refcounted_stack_init(int size)
+{
+	gc_refcounted_stack *stack;
+	stack = (gc_refcounted_stack*) emalloc(sizeof(gc_refcounted_stack));
+
+	stack->count = 0;
+	stack->max = size;
+	stack->bottom = (zend_refcounted**) emalloc(stack->max * sizeof(zend_refcounted*));
+
+	return stack;
+}
+
+static void gc_refcounted_stack_push(gc_refcounted_stack *stack, zend_refcounted *ref)
+{
+	if (stack->count >= stack->max) { // if max capacity is full, double in size
+		stack->max *= 2;
+		stack->bottom = (zend_refcounted**) erealloc(stack->bottom, stack->max * sizeof(zend_refcounted*));
+	}
+
+	*(stack->bottom + stack->count) = ref;
```

Comment:
A more obvious way to write this is `stack->bottom[stack->count] = ref` (same below).

Reply:
Yes, it's more readable.